packages/hardhat-core/src/internal/hardhat-network/provider/node.ts

```diff
@@ -2297,6 +2325,37 @@ Hardhat Network's forking functionality only works with blocks from at least spu
 
     return { maxFeePerGas, maxPriorityFeePerGas };
   }
+
+  /** search this._hardforkActivationBlocks for the highest block number that
+   * isn't higher than blockNum, and then return that found block number's
+   * associated hardfork name. */
+  private _selectHardforkFromActivations(
+    blockNum: BN
+  ): typeof HARDHAT_NETWORK_SUPPORTED_HARDFORKS[number] {
+    let highestFound: {
+      hardfork: typeof HARDHAT_NETWORK_SUPPORTED_HARDFORKS[number];
+      block: number;
+    } = { hardfork: "", block: 0 };
+    Object.entries(this._hardforkActivationBlocks).forEach((entry) => {
+      const [hardfork, block] = entry;
+      if (!HARDHAT_NETWORK_SUPPORTED_HARDFORKS.includes(hardfork)) {
```

Comment:
Can move this to Hardhat Runner's config validation logic. At this level of abstraction we assume everything has already been validated.